### PR TITLE
Spending analytics servlet

### PR DESCRIPTION
**Pull request by emilyvera**
Add servlet to compute user analytics using SpendingAnalytics helper class. Again, this only computes analytics by store name, but categories may be added later

### CODE REVIEW

File: src/test/java/SpendingAnalyticsServletTest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+  @Test
+  public void doGetWithReceiptsInDatastore() throws IOException {
+    // Receipts in datastore:
+    // Walmart: $26.12, Contoso: $14.51, Main Street Restaurant: $29.01
+    TestUtils.addTestReceipts(datastore);
+
+    // Perform doGet.
+    servlet.doGet(request, response);
+    writer.flush();
```
Comment (basaam0) [inline src/test/java/SpendingAnalyticsServletTest.java:77]: I don't think this comment is necessary,

File: src/test/java/SpendingAnalyticsServletTest.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void doGetWithReceiptsInDatastore() throws IOException {
+    // Receipts in datastore:
+    // Walmart: $26.12, Contoso: $14.51, Main Street Restaurant: $29.01
+    TestUtils.addTestReceipts(datastore);
```
Comment (acheng1) [inline src/test/java/SpendingAnalyticsServletTest.java:73]: nit: similar comment about making sure that a test can be understood without looking at the definitions of helper method...

Comment (emilyvera) [reply src/test/java/SpendingAnalyticsServletTest.java:73]: The comments on lines 71/72 are supposed to show what the test is without all the clutter of the setup. Do you think I should still change these tests?

Comment (acheng1) [reply src/test/java/SpendingAnalyticsServletTest.java:73]: Yeah, generally we would prefer code that is self explanatory over adding comments.

Comment (emilyvera) [reply src/test/java/SpendingAnalyticsServletTest.java:73]: thanks that makes sense

**Pull request merged**